src/lib/controller/message_handler.ts

```diff
@@ -118,3 +153,60 @@ export class ControllerMessageHandler {
     }
   }
 }
+
+function processInclusionOptions(
+  clientsController: ClientsController,
+  client: Client,
+  message: IncomingMessageController
+): InclusionOptions | ReplaceNodeOptions {
+  if ("options" in message) {
+    if (
+      message.options.strategy === InclusionStrategy.Default ||
+      message.options.strategy === InclusionStrategy.Security_S2
+    ) {
+      message.options.userCallbacks = {
+        grantSecurityClasses: async (
+          requested: InclusionGrant
+        ): Promise<InclusionGrant | false> => {
+          clientsController.grantSecurityClassesPromise = createDeferredPromise<
+            InclusionGrant | false
```

Comment:
I don't think you need to specify the type - it should be inferred from `clientsController.grantSecurityClassesPromise`